server/metric_reporter.go

```diff
@@ -111,20 +117,31 @@ func (m *MetricReporter) Register(metricName metric.Name, collector metric.Colle
 }
 
 // Identify will identify the workspace and update the subscription plan.
-func (m *MetricReporter) identify() {
+func (m *MetricReporter) identify(ctx context.Context) {
 	plan := api.FREE.String()
 	orgID := ""
 
 	if m.subscription != nil {
 		plan = m.subscription.Plan.String()
 		orgID = m.subscription.OrgID
 	}
+
+	principal, err := m.store.GetPrincipalByID(ctx, principalIDForFirstUser)
+	if err != nil {
+		log.Debug("get principal by id failed", zap.Int("id", principalIDForFirstUser), zap.Error(err))
+	}
+	email := ""
+	if principal != nil {
+		email = principal.Email
+	}
+
 	if err := m.reporter.Identify(&metric.Identifier{
 		ID: m.workspaceID,
 		Labels: map[string]string{
 			identifyTraitForPlan:    plan,
 			identifyTraitForVersion: m.version,
 			identifyTraitForOrgID:   orgID,
+			identifyTraitForEmail:   email,
```

Comment:
Instead of using a label, shall we use the email field from trait?

https://segment.com/docs/connections/sources/catalog/libraries/server/go/quickstart/

Reply:
They're actually the same for us, as the segment go package will mark the trait as a map, and set email as a label.
But sure, I can update the code to use the `SetEmail` interface to make it more clear.

Reply:
I see, thanks! If it works, it works.